File: app/src/main/java/ru/kireev/mir/laregistrationmain/data/Volunteer.java
package ru.kireev.mir.laregistrationmain.data;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "volunteers")
public class Volunteer {

    @PrimaryKey(autoGenerate = true)
    private int uniqueId;
    private int index;
    private String name;
    private String surname;
    private String callSign;
    private String phoneNumber;
    private String isSent;

    public Volunteer(int uniqueId, int index, String name, String surname, String callSign, String phoneNumber, String isSent) {
        this.uniqueId = uniqueId;
        this.index = index;
        this.name = name;
        this.surname = surname;
        this.callSign = callSign;
        this.phoneNumber = phoneNumber;
        this.isSent = isSent;
    }

    @Ignore
    public Volunteer(int index, String name, String surname, String callSign, String phoneNumber, String isSent) {
        this.index = index;
        this.name = name;
        this.surname = surname;
        this.callSign = callSign;
        this.phoneNumber = phoneNumber;
        this.isSent = isSent;
    }

    public String isSent() {
        return isSent;
    }

    public void setSent(String sent) {
        isSent = sent;
    }

    public int getUniqueId() {
        return uniqueId;
    }

    public void setUniqueId(int uniqueId) {
        this.uniqueId = uniqueId;
    }

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public String getCallSign() {
        return callSign;
    }

    public void setCallSign(String callSign) {
        this.callSign = callSign;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }
}

File: app/src/main/java/ru/kireev/mir/registrarlizaalert/data/database/entity/Volunteer.kt
package ru.kireev.mir.registrarlizaalert.data.database.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import ru.kireev.mir.registrarlizaalert.domain.model.VolunteerStatus

@Entity(tableName = "volunteers",
        foreignKeys = [
            ForeignKey(
                    entity = Group::class, parentColumns = arrayOf("id"),
                    childColumns = arrayOf("groupId"),
                    onDelete = ForeignKey.SET_NULL
            )
        ])
data class Volunteer(
    @PrimaryKey(autoGenerate = true)
        var uniqueId: Int,
    var _index: Int,
    var fullName: String,
    var callSign: String = "",
    var nickName: String = "",
    var region: String = "",
    var phoneNumber: String,
    var car: String = "",
    var isSent: String = "false",
    var status: VolunteerStatus = VolunteerStatus.ACTIVE,
    var notifyThatLeft: String = "false",
    var timeForSearch: String = "",
    var groupId: Int? = null
) {
    override fun toString(): String {
        return "$fullName ($callSign)"
    }
}

fun emptyVolunteer() = Volunteer(0, 0, "", phoneNumber = "")